src/main/java/de/tum/in/www1/artemis/repository/SubmissionRepository.java

```diff
@@ -450,25 +450,14 @@ SELECT COUNT(DISTINCT s)
      * @return a new submission for the given type connected to the given participation
      */
     default Submission initializeSubmission(Participation participation, Exercise exercise, SubmissionType submissionType) {
```

Comment:
same here. you could think about sealing the submissions. this should remove the default case

Reply:
Actually here we would need to seal the Exercise, not the Submission, but it wouldn't give us too much, because we would still need the `null` case anyways.

Reply:
Yes, but not the default case.